Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -0,0 +1,280 @@
+    pp_amr.query("max_level", nLevel);
+    nLevel += 1;
+
+    amrex::ParmParse pp_rd_name(rd_name);
+    pp_rd_name.query("x_probe", x_probe);
+    pp_rd_name.query("y_probe", y_probe);
+    pp_rd_name.query("z_probe", z_probe);
+
+    const bool raw_specified = pp_rd_name.query("raw_fields", raw_fields);
+    if(!raw_specified) raw_fields = false;
+
+    const bool interp_order_specified = pp_rd_name.query("raw_fields", interp_order);
+    if(!interp_order_specified) interp_order = 1;
+
+    // resize data array
+    m_data.resize(noutputs*nLevel, 0.0_rt);
+
+    if (ParallelDescriptor::IOProcessor())
+    {
+        if ( m_IsNotRestart )
+        {
+            // open file
+            std::ofstream ofs{m_path + m_rd_name + "." + m_extension, std::ofstream::out};
+            // write header row
+            int c = 0;
+            ofs << "#";
+            ofs << "[" << c++ << "]step()";
+            ofs << m_sep;
+            ofs << "[" << c++ << "]time(s)";
+            for (int lev = 0; lev < nLevel; ++lev)
+            {
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Ex_lev" + std::to_string(lev) + " (V/m)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Ey_lev" + std::to_string(lev) + " (V/m)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Ez_lev" + std::to_string(lev) + " (V/m)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_|E|_lev" + std::to_string(lev) + " (V/m)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Bx_lev" + std::to_string(lev) + " (T)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_By_lev" + std::to_string(lev) + " (T)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Bz_lev" + std::to_string(lev) + " (T)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_|B|_lev" + std::to_string(lev) + " (T)";
+            }
+            ofs << std::endl;
+            // close file
+            ofs.close();
+        }
+    }
+}
+// end constructor
+
+// function that computes maximum field values
+void FieldProbe::ComputeDiags (int step)
+{
+    // Judge if the diags should be done
+    if (!m_intervals.contains(step+1)) { return; }
+
+    // get a reference to WarpX instance
+    auto & warpx = WarpX::GetInstance();
+
+    // get number of level
+    const auto nLevel = warpx.finestLevel() + 1;
+
+    // loop over refinement levels
+    for (int lev = 0; lev < nLevel; ++lev)
+    {
+
+        const amrex::Geometry& gm = warpx.Geom(lev);
+        const auto prob_lo = gm.ProbLo();
+        const auto prob_hi = gm.ProbHi();
+
+        const bool probe_in_domain = x_probe >= prob_lo[0] and y_probe >= prob_lo[1] and
+                                     z_probe >= prob_lo[2] and x_probe <= prob_hi[0] and
+                                     y_probe <= prob_hi[1] and z_probe <= prob_hi[2];
```

Comment:
This probably does not work in 2D, where `z_probe` should be compared to `probe_lo/hi[1]` (`probe_lo/hi[2]` does not exist) and `y_probe` is irrelevant (probably safer to not declare it at all). You can have a look at how it's done in `FieldReduction.H` for an example.